### PR TITLE
Configuration for DocumentWriter can take a client.

### PR DESCRIPTION
**Pull request by andricDu**

### CODE REVIEW

**Comment by andricDu** [comment]
@slovit @btiernay

File: dcc-common-es/src/main/java/org/icgc/dcc/dcc/common/es/DocumentWriterFactory.java [hunk 1]
```diff
@@ -49,7 +49,7 @@
   private static final int BULK_ACTIONS = -1; // Unlimited
 
   public static DocumentWriter createDocumentWriter(DocumentWriterConfiguration configuration) {
-    val client = newTransportClient(configuration.esUrl(), true);
+    val client = configuration.client != null ? configuration.client : newTransportClient(configuration.esUrl(), true);
```
Comment (btiernay) [inline dcc-common-es/src/main/java/org/icgc/dcc/dcc/common/es/DocumentWriterFactory.java]: Use getters please. We like encapsulation in this family :)

File: dcc-common-es/src/main/java/org/icgc/dcc/dcc/common/es/DocumentWriterFactory.java [hunk 2]
```diff
@@ -59,7 +59,7 @@ public static DocumentWriter createDocumentWriter(DocumentWriterConfiguration co
   }
 
   @SuppressWarnings("resource")
-  private static TransportClient newTransportClient(@NonNull String esUri, boolean sniff) {
+  private static Client newTransportClient(@NonNull String esUri, boolean sniff) {
```
Comment (btiernay) [inline dcc-common-es/src/main/java/org/icgc/dcc/dcc/common/es/DocumentWriterFactory.java]: `newClient`

**Review by btiernay — CHANGES_REQUESTED** [review]
Do it

**Comment by andricDu** [comment]
ok to test

**Pull request merged**